packages/twenty-server/src/core/auth/auth.resolver.ts

```diff
@@ -128,6 +129,23 @@ export class AuthResolver {
     return result;
   }
 
+  @Mutation(() => Verify)
+  @UseGuards(JwtAuthGuard)
+  async generateJWT(@Args() args: GenerateJwtInput): Promise<Verify> {
+    const workspace = await this.workspaceRepository.findOneBy({
+      id: args.workspaceId,
+    });
+
+    assert(workspace, 'workspace doesnt exist', NotFoundException);
+
+    const token = await this.tokenService.generateSwitchWorkspaceToken(
```

Comment:
before generating, you need to make sure that the authenticated user belongs to the workspace related to `args.workspaceId`, otherwise every user could log to every workspace

Reply:
You're correct! Great catch